easybuild/tools/github.py

```diff
@@ -332,14 +334,32 @@ def fetch_easyconfigs_from_pr(pr, path=None, github_user=None):
         raise EasyBuildError("Failed to get data for PR #%d from %s/%s (status: %d %s)",
                              pr, GITHUB_EB_MAIN, GITHUB_EASYCONFIGS_REPO, status, pr_data)
 
-    # 'clean' on successful (or missing) test, 'unstable' on failed tests
+    dl_dev = False
+    # if PR is open and mergable, download develop and patch
     stable = pr_data['mergeable_state'] == GITHUB_MERGEABLE_STATE_CLEAN
+    closed = pr_data['state'] == GITHUB_STATE_CLOSED and pr_data['merged'] == False
+    # 'clean' on successful (or missing) test, 'unstable' on failed tests
     if not stable:
         _log.warning("Mergeable state for PR #%d is not '%s': %s.",
                      pr, GITHUB_MERGEABLE_STATE_CLEAN, pr_data['mergeable_state'])
 
-    for key, val in sorted(pr_data.items()):
-        _log.debug("\n%s:\n\n%s\n" % (key, val))
+    if stable and not closed:
+        # wether merged or not, download develop
+        develop = download_repo(repo=GITHUB_EASYCONFIGS_REPO, branch='develop', path=path)
+        if not os.path.isdir(develop):
+            raise EasyBuildError("Downloading of develop branch failed: not found in %s", develop)
+        dl_dev = True
+
+        # not merged: download patch
+        if not pr_data['merged']:
+            patch_name = 'patch%s' % pr
+            patch_url = URL_SEPARATOR.join([GITHUB_URL, GITHUB_EB_MAIN, GITHUB_EASYCONFIGS_REPO, GITHUB_PR, '%s.patch' % pr])
+            patch_path = os.path.join(path, patch_name)
+            patch = download_file(patch_name, patch_url, patch_path)
+            if patch is None:
+                raise EasyBuildError("Failed to download file %s to %s", patch_url, patch_path)
```

Comment:
we're already downloading the patch via `diff_url` below, avoid duplicate code please :)

and use `pr_date['diff_url']` rather than composing the patch URL yourself, i.e. use the GitHub API :)